database/migrations/default/1736197283308_jurisdiction_refactor/up.sql

```diff
@@ -0,0 +1,143 @@
+alter table geo.atd_jurisdictions add column jurisdictions_id integer;
+
+create or replace function public.crashes_set_spatial_attributes()
+returns trigger
+language plpgsql
+as $function$
+begin
+    if (new.latitude is not null and new.longitude is not null) then
+        -- save lat/lon into geometry col
+        new.position = st_setsrid(st_makepoint(new.longitude, new.latitude), 4326);
+        --
+        -- get location polygon id
+        --
+        if (new.rpt_road_part_id != 2 and upper(ltrim(new.rpt_hwy_num)) in ('35', '183','183A','1','290','71','360','620','45','130')) then
+            -- use level 5 polygon
+            new.location_id = (
+                select
+                    location_id
+                from
+                    public.atd_txdot_locations
+                where
+                    location_group = 2 -- level 5
+                    and st_contains(geometry, new.position)
+                limit 1);
+        else
+            -- use the other polygons
+            new.location_id = (
+                select
+                    location_id
+                from
+                    public.atd_txdot_locations
+                where
+                    location_group = 1 -- not level 5
+                    and st_contains(geometry, new.position)
+                limit 1);
+        end if;
+
+        raise debug 'found location: % compared to previous location: %', new.location_id, old.location_id;
+        --
+        -- check if in austin full purpose jurisdiction
+        --
+        new.in_austin_full_purpose = EXISTS (
+            select 1 
+            from geo.atd_jurisdictions 
+            where JURISDICTION_LABEL = 'AUSTIN FULL PURPOSE' AND  ST_Intersects(geometry, new.position)
+        );
```

Comment:
here's the updated code. we're filtering on the jurisdiction label rather than the arbitrary ID sequence

Reply:
👋 goodbye magic number - thank you!